client/src/context/gameContext.js

```diff
@@ -49,11 +55,41 @@ export const GameProvider = (props) => {
   };
 
   const cardUpdate = (data) => {
-    // console.log('listening');
+    const updatedCards = cardsRef.current;
```

Comment:
Generally in React, we want to avoid mutating state directly. `slice()` returns a cloned array; `const updatedCards = cardsref.current.slice();` .